static/vue.js/data_collection_frequency.js

```diff
@@ -16,7 +16,7 @@ Vue.component('modal', {
       modalHeader:'Add Data Collection Frequency',
     }, 
     beforeMount: function(){
-        axios.get('/indicators/data_collection_frequency/list')
+        this.makeRequest('GET', '/indicators/data_collection_frequency/list')
```

Comment:
Could we use `async/await` syntax so that we can avoid callbacks.

Reply:
@michaelbukachi by call-backs you mean promises? Sure, good point, for a small implementation on the same server, without any network lag, it makes no difference. May be code-wise, which only changes when you have nested promises, then at that point await/async is cleaner.